src/_pytest/mark/structures.py

```diff
@@ -368,6 +369,24 @@ def get_unpacked_marks(obj) -> List[Mark]:
     return normalize_mark_list(mark_list)
 
 
+MARKERS_BY_CLASS_NAME: Dict[str, List[Mark]] = {}
+
+
+def get_all_mro_marks(cls, module_name):
+    all_markers = getattr(cls, "pytestmark", [])
+    if cls is None:
+        return all_markers
+    if not isinstance(all_markers, list):
+        all_markers = [all_markers]
+    for parent_obj in cls.__mro__[::-1]:
+        full_class_name = f"{module_name}_{parent_obj.__name__}"
+        if not MARKERS_BY_CLASS_NAME.get(full_class_name):
+            MARKERS_BY_CLASS_NAME[full_class_name] = get_unpacked_marks(parent_obj)
```

Comment:
going for the class name here is potentially error prone i would strongly suggest lookup by the type directly

additionally it is incorrect to use get_unpacked_marks as that would unconditionally fetch markers of base classes
(if identity de-duplication was implemented that would be a temporaryly acceptable shortcoming)

Reply:
Done